mat/diagonal_test.go

```diff
@@ -458,3 +460,11 @@ func TestDiagonalAtSet(t *testing.T) {
 		}
 	}
 }
+
+func randDiagDense(size int) *DiagDense {
+	t := NewDiagDense(size, nil)
+	for i := 0; i < size; i++ {
+		t.SetDiag(i, rand.Float64())
```

Comment:
We don't use the package-level rand functions since it makes tests non-reproducible. You can make this function take a `rand.Source` and create a `rand.Rand` to use.